add post

maelle opened the pull request.

maelle commented:
Direct link to post preview https://deploy-preview-244--ropensci.netlify.com/blog/2018/08/28/birds-ocr/

maelle commented:
I don't understand the absence of line breaks before some titles. 😿

jeroen commented:
```
> spelling::spell_check_files("2018-08-28-birds-ocr.md")
  WORD         FOUND IN
australia    2018-08-28-birds-ocr.md:367
BHL          2018-08-28-birds-ocr.md:336,348
cld          2018-08-28-birds-ocr.md:12,13
flickr       2018-08-28-birds-ocr.md:34
Flickr       2018-08-28-birds-ocr.md:55
github       2018-08-28-birds-ocr.md:15
Jeroen       2018-08-28-birds-ocr.md:39
Maëlle      2018-08-28-birds-ocr.md:5
magick       2018-08-28-birds-ocr.md:10
md           2018-08-28-birds-ocr.md:15
ocr          2018-08-28-birds-ocr.md:2
Ooms’      2018-08-28-birds-ocr.md:39
OSD          2018-08-28-birds-ocr.md:132
paclage      2018-08-28-birds-ocr.md:350
Radolfzell   2018-08-28-birds-ocr.md:8
rOpenSci     2018-08-28-birds-ocr.md:315
rOpenSci’s 2018-08-28-birds-ocr.md:22,39,231,348
taxize       2018-08-28-birds-ocr.md:14
tesseract    2018-08-28-birds-ocr.md:11
Tesseract    2018-08-28-birds-ocr.md:122,128
th           2018-08-28-birds-ocr.md:8
toolbelt     2018-08-28-birds-ocr.md:211
wll          2018-08-28-birds-ocr.md:5
worfklow     2018-08-28-birds-ocr.md:342
yaml         2018-08-28-birds-ocr.md:15
```

jeroen commented:
> I don't understand the absence of line breaks before some titles. 😿
Hmmm I'll look into that as well.

maelle commented:
Thanks, by the way could your spelling code become a part of all PRs?

jeroen commented:
I usually avoid using two headers right after one another. This is a bit of a personal preference, but I think you can omit the second header ("image preperation") here:

<img width="804" alt="screen shot 2018-08-08 at 4 13 59 pm" src="https://user-images.githubusercontent.com/216319/43842681-27ac65e8-9b26-11e8-9c7f-dd2fa3105d77.png">

maelle commented:
@jeroen Thanks, I removed the typos, and added text between headers.

I have tried using one line break instead of two but it didn't work, some headers still appear on the line before... I can't see any difference between them and other, correctly-placed, headers! 🙀

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+
+I wanted to try to extract the bird names on images using packages of
+Jeroen Ooms’, rOpenSci’s post-doc hacker & C(++)-bindings wizard. In
+particular, I wanted to try extract names of
```
s/try extract names of magick/try to extract names using magick/ ?

maelle replied:
rephrased

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+[downloaded all pictures from one of the Birds of Australia
+albums](https://www.flickr.com/photos/biodivlibrary/sets/72157694191194992).
+
+I wanted to try to extract the bird names on images using packages of
```
s/names on images/names from images/ ?

maelle replied:
done

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+taxonomy magic by Scott Chamberlain, one of rOpenSci’s co-founders, to
+resolve the names extracted.
+
+### OCR-bird naming workflow, piece by piece
```
omit hyphen
looks like it's "OCR-bird" though I know that's not what you mean

stefaniebutland replied:
same for heading below

maelle replied:
👍

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+rOpenSci’s packages binding powerful C++ libraries and open taxonomy
+data, how much information can we automatically extract from images?
+Maybe not much, but we’ll at least have explored image manipulation,
+optical character recognition, language detection, taxonomic name
```
add (OCR) since abbrev is used in heading below without definition
search doc and use OCR where appropriate

maelle replied:
good point

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+
+### OCR-bird naming workflow, piece by piece
+
+In this section, we explain the different elements of our R workflow:
```
mixing of "I" and "we" confusing

maelle replied:
I'm rephrasing a bit but not sure I made the choice in the right direction of changes.

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
````diff
@@ -0,0 +1,411 @@
+```
+
+Each image shows a bird and its name. Images are either landscape- or
+portrait- oriented, but for the sake of simplicity, I’ll act as if they
````
s/portrait- oriented/portrait-oriented/
in 2 places

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+Each image shows a bird and its name. Images are either landscape- or
+portrait- oriented, but for the sake of simplicity, I’ll act as if they
+were all portrait- oriented. *What I’d do if I weren’t after simplicity
+but still as lazy would be to duplicate images rotated
```
revised for reader clarity:
*What I’d do if I weren’t after simplicity
but still as lazy, would be to duplicate images rotated
(`magick::image_rotate`) in all possible directions and apply the
workflow, hoping to get good data from one of the rotated versions.*

maelle replied:
rephrased

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+<img src="/img/blog-images/2018-08-28-birds-ocr/unnamed-chunk-3-1.png" width="1376" />
+
+It does look cleaner now!
+
```
need line break before heading (I know these are trouble)

maelle replied:
whyyy are they so much trouble?! I didn't succeed in getting them right :sob:

maelle replied:
I tried `<br>` and before that I had tried comparing them to working headers, no luck.

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+mode](https://github.com/tesseract-ocr/tesseract/wiki/ImproveQuality#page-segmentation-method)
+choosing 1 for “Automatic page segmentation with Orientation and script
+detection (OSD)”. We tried using Latin instead of English training data,
+but the results were less good.
```
s/less/not as/

maelle replied:
👍

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+to image, or `magick::image_ocr` for a `magick` object which is quite
+handy in our pipeline.
+
+The function below also filter results of the OCR using language
```
s/filter/filters/

maelle replied:
✔️

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+
+The function below also filter results of the OCR using language
+detection. By only keeping text recognized as either Latin or English by
+one of the two languages detection packages `cld2` and `cld3` that are
```
s/languages detection/language detection/

maelle replied:
✔️

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+    ## 1 climacteris picumnus <NA>  la    birds/n115_w1150_42399797481_o.jpg
+    ## 2 brown tree creeper   en    <NA>  birds/n115_w1150_42399797481_o.jpg
+
+So we see that the image transformation was quite useful.
```
Sentence "So we see that the image transformation was quite useful." is a bit unclear after a long bunch of code snippets i.e. where does a novice coder look? Could you move that line above the tibble? or add a phrase giving example of "climacteris picumnus"?

maelle replied:
I split the chunk and added more comments, is it enough?

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+
+#### Taxonomic name resolution
+
+The [`taxize` package](https://github.com/ropensci/taxize/) by Scott
```
with Zachary Foster now having maintainer status too, should he be mentioned even though Scott is major committer?

maelle replied:
👍

maelle replied:
Ah no I re-checked taxize's maintainer is @sckott, it's `taxa` that got a new maintainer.

maelle replied:
np, it was a good remark

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+#### Taxonomic name resolution
+
+The [`taxize` package](https://github.com/ropensci/taxize/) by Scott
+Chamberlain, taxonomic toolbelt for R, provides access to many fantastic
```
maybe rephrase so Scott doesn't interrupt the package name and it's purpose:

"The [`taxize` package](https://github.com/ropensci/taxize/), a taxonomic toolbelt for R, by Scott Chamberlain..."

or "The [`taxize` package(https://github.com/ropensci/taxize/) by Scott Chamberlain, is a taxonomic toolbelt for R, providing access to many fantastic data sources and tools for taxonomy"

maelle replied:
✅

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+Chamberlain, taxonomic toolbelt for R, provides access to many fantastic
+data sources and tools for taxonomy. One of them, the Global Name
+Resolver, provides, well, resolution of taxonomic names, sadly not
+common names. `taxize::gnr_resolve` has many options, we only make use
```
s/options, we /options, but we /

maelle replied:
re-phrased

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+    ## 1 climacteris picum~ Climacteris pi~ Climacteris p~ NCBI             0.988
+
+English names could be cleaned up a bit using the [`spelling`
+package](https://github.com/ropensci/spelling), also an rOpenSci’s
```
s/an rOpenSci’s package/an rOpenSci package/

maelle replied:
✅

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+image processing and OCR tweaking steps! Maybe one could frame the
+parameter search [as a machine learning
+problem](https://twitter.com/dmi3k/status/1024919690768990209). We
+should also keep in mind that such Natural History images are quite hard
```
no caps in Natural History?

maelle replied:
✅

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+to start with. A better idea here might be to get a full list of names
+of Australian birds, potentially leveraging [the `taxizedb` package by
+Scott Chamberlain](https://github.com/ropensci/taxizedb), and to then
+fuzzy-match them with the names we have.
```
you should comment on whether that was helpful or not

maelle replied:
I can't because it's just a random idea I wrote. Will add "untested".

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+Global Name Resolver](https://resolver.globalnames.org/about). Discover
+more packages from our suite [here](https://ropensci.org/packages/).
+
+#### Applicability of this OCR bird-naming workflow
```
different hyphen use here. please resolve

maelle replied:
✅

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+    for language detection.
+
+We also used [a function from
+`taxize`](https://github.com/ropensci/taxize) allowing to use [the
```
s/allowing to use/allowing us to use/ ?

maelle replied:
✅

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+for its collection, see [this
+example](https://www.biodiversitylibrary.org/item/250938#page/37/mode/1up).
+I wasn’t able to find information about the software powering this OCR.
+What I *was* able to find out is that the BHL uses [purposeful
```
define abbrev BHL above and use consistently below if you use it >= 3x

maelle replied:
✅  except in a header of the conclusion to make it easier to skim

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+
+#### More data from the Biodiversity Heritage Library
+
+If you’re interested in the data, images but not only, of the
```
suggested rephrase:

If you’re interested in other types of data from the BHL, in addition to the images,

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+    ## [5] "The birds of Australia,"                                                                 
+    ## [6] "The birds of Australia,"
+
+And to get the OCR results of the pages of the book we used, we could
```
can you rephrase toclarify whether you are looking for page numbers?

maelle replied:
I added comments in the code chunk

stefaniebutland commented on content/blog/2018-08-28-birds-ocr.md:
```diff
@@ -0,0 +1,411 @@
+    ## [6] "49 6 \nH \n(jronvolcl. del. \nN E O SIT TA LE CJ C O CE PHAI.A. \n( WHITE ¦ HE AID EE THE EH UN HE FL). \nNEOSHTA ALBATA \n(F IE E> T Ft E EE UNNEFlj. \nNEOSITTA CHRYSOPTERA \nf OFi. A. NGE - wing-e d tree runner). \nWitWLjA \n"
+
+So there’s quite a lot to explore!
+
```
same line break problem

maelle commented:
Many thanks @stefaniebutland , super useful as usual.

I was unable to fix headers. @jeroen do you have any idea why the headers

* Text extraction
* "More birding soon!" at the very end

don't start on a new line?! I added `<br>`, and had tested other small things earlier, with no success.

jeroen commented:
It looks like displaying `article h4` headers as inline has been a [style choice](https://github.com/ropensci/roweb2/blob/master/themes/ropensci/static/css/sections.pcss#L265) by the designer. Maybe we should just remove that?

maelle commented:
Maybe I'm being lazy but I think it could be removed, why would one want inline headers?

jeroen commented:
@maelle I agree, I'm just going to remove that.

jeroen commented:
Looks good to me, cool showcase of combination of ropensci packages.

maelle commented:
Thanks @jeroen! Then after my vacation I'll "just" need to rebase the commit 😹
